### PR TITLE
Custom page design tweaks + read manager refactor

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-2506)
[this task](https://uktrade.atlassian.net/browse/ED-2519)
[this task](https://uktrade.atlassian.net/browse/ED-2530)

image

### CODE REVIEW

File: article/helpers.py [hunk 1]
```diff
@@ -59,24 +59,22 @@ def __init__(self, request):
         self.request = request
 
     persist_article = abc.abstractproperty()
-    retrieve_articles = abc.abstractproperty()
+    retrieve_article_uuids = abc.abstractproperty()
```
Comment (konradko) [inline article/helpers.py]: not `abstractmethod`?

File: article/helpers.py [hunk 2]
```diff
@@ -86,6 +84,7 @@ def remaining_reading_time_in_group(self, group_name):
         """ Return the remaining reading time in minutes
             for unread articles in the group
         """
+
         read_articles_uuids = self.read_articles_keys_in_group(group_name)
```
Comment (sdonk) [inline article/helpers.py]: It's not relevant to your code, but I'd change this to `read_articles_uuids_in_group` for consistency

File: core/tests/test_core_middleware.py [hunk 3]
```diff
@@ -4,11 +4,11 @@
 
 from django.http import HttpResponse
 
-from ui import middleware
```
Comment (konradko) [inline core/tests/test_core_middleware.py]: don't think we need `test_core_middleware.py` name, it's already under `core` folder

File: article/helpers.py (hunk 1 above)
Comment (richtier) [reply article/helpers.py]: ah yes, it should be

File: core/tests/test_core_middleware.py (hunk 3 above)
Comment (richtier) [reply core/tests/test_core_middleware.py]: pytest sometimes gets confused when it encounters test files with the same name and something like "tried to import blah but __file__ is not what we expected"

**Comment by richtier** [comment]
The failing code cov is code that will never run - inside an abstract method.

**Pull request merged**